hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -1125,6 +1130,14 @@ private HoodieData<HoodieRecord> getFilesPartitionRecords(String createInstantTi
     return filesPartitionRecords.union(fileListRecords);
   }
 
+  protected void closeInternal() {
+    try {
+      close();
+    } catch (Exception e) {
```

Comment:
If we do not want `#close` to throw checked exception everywhere, just remote the throws from the interface.

Reply:
AutoCloseable would throw an exception if the resource is not closed. However, I think it is better to catch and wrap in HoodieException. Easier to search and debug.